tools/backup/templates/backup.yaml

```diff
@@ -1,60 +1,77 @@
-          imagePullPolicy: Always
-          env:
-          # Addr must include a host and port, but no scheme.  See the --from parameter here:
-          # See https://neo4j.com/docs/operations-manual/current/backup/performing/
-          - name: NEO4J_ADDR
-            value: {{ .Values.neo4jaddr }}
-          - name: BUCKET
-            value: {{ .Values.bucket }}
-          - name: DATABASE
-            value: {{ .Values.database }}
-          - name: GOOGLE_APPLICATION_CREDENTIALS
-            value: {{ .Values.credentials }}
-          - name: HEAP_SIZE
-            value: {{ .Values.heapSize }}
-          - name: PAGE_CACHE
-            value: {{ .Values.pageCache }}
-          - name: FALLBACK_TO_FULL
-            value: "{{ .Values.fallbackToFull }}"
-          - name: CHECK_CONSISTENCY
-            value: "{{ .Values.checkConsistency }}"
-          - name: CHECK_INDEXES
-            value: "{{ .Values.checkIndexes }}"
-          - name: CHECK_GRAPH
-            value: "{{ .Values.checkGraph }}"
-          - name: CHECK_LABEL_SCAN_STORE
-            value: "{{ .Values.checkLabelScanStore }}"
-          - name: CHECK_PROPERTY_OWNERS
-            value: "{{ .Values.checkPropertyOwners }}"
-          volumeMounts:
-          - name: creds
-            mountPath: /auth
-            readOnly: true
-      volumes:
-      - name: creds
-        secret:
-          secretName: {{ .Values.secretName }}
-          items:
-          - key: credentials.json
-            path: credentials.json
+      template:
+        spec:
+          restartPolicy: Never
+          containers:
+            - name: graph-backup
+              image: {{ .Values.image }}:{{ .Values.imageTag }}
+              imagePullPolicy: Always
+              env:
+                - name: NEO4J_ADDR
+                  value: {{ .Values.neo4jaddr }}
+                - name: DATABASE
+                  value: {{ .Values.database }}
+                - name: CLOUD_PROVIDER
+                  value: {{ .Values.cloudProvider }}
+                - name: BUCKET
+                  value: {{ .Values.bucket }}
+                - name: HEAP_SIZE
+                  value: {{ .Values.heapSize }}
+                - name: PAGE_CACHE
+                  value: {{ .Values.pageCache }}
+                - name: FALLBACK_TO_FULL
+                  value: "{{ .Values.fallbackToFull }}"
+                - name: CHECK_CONSISTENCY
+                  value: "{{ .Values.checkConsistency }}"
+                - name: CHECK_INDEXES
+                  value: "{{ .Values.checkIndexes }}"
+                - name: CHECK_GRAPH
+                  value: "{{ .Values.checkGraph }}"
+                - name: CHECK_LABEL_SCAN_STORE
+                  value: "{{ .Values.checkLabelScanStore }}"
+                - name: CHECK_PROPERTY_OWNERS
+                  value: "{{ .Values.checkPropertyOwners }}"
+              volumeMounts:
+                {{ if eq .Values.cloudProvider "aws" }}
```

Comment:
in the blocks where you specify a volumeMount and volumes, you don't need to switch on aws vs. gcp, it causes code to be repeated.  No matter which is used, you could just use one; `/credentials/secret` and the content of it would be the secret content, either way GCP or AWS flag just controls how that secret data is interpreted by the CLI

(The reason I'm sensitive to code being repeated here is that probably in 90 days someone will come along and ask for the same thing, but Azure too.  ;)